2110336-2565-2/sec33_Group4_BookBix_Backend
# Feature/payment

## Description

- This pull request adds functionality to create a Stripe product and price in a single API call using the `createProductAndPrice` method. A new route is also included to handle the `POST` requests for creating a new product and price in a Stripe connected account.
- This pull request also adds functionality to create a Stripe coupon in a single API call using the createCoupon method. A new route is included to handle the POST requests for creating a new coupon in a Stripe connected account.

## Changes

- Added a `createProductAndPrice` method to the `StripeService` to create a new product and price in a Stripe connected account using a single API call.
- Added a new `POST /stripe/create-product` route to handle requests to create a new product and price in a Stripe connected account.
- Included an example JSON payload to use when testing the `POST /stripe/create-product` route in Postman.
- Added a createCoupon method to the StripeService to create a new coupon in a Stripe connected account using a single API call.
- Added a new POST /stripe/create-coupon route to handle requests to create a new coupon in a Stripe connected account.
Included parameters such as percentOff, duration, durationInMonths, and productIds in the request body to create a coupon.
Returned the created coupon as a response in the JSON format.

## Testing

To test the createProductAndPrice, follow these steps:

1. Start the server by running `npm start`.
2. Open Postman and use request to `{{URL}}/stripe/create-product`. (named "create-product")
3. Set the request method to `POST`.
4. Copy the example JSON payload below and paste it into the request body.
5. Click the "Send" button to send the request to the server.
6. Verify that the response includes the newly created product and price in the Stripe connected account.

To test the create-coupon, follow these steps:
1. Open Postman and use request to `{{URL}}/stripe/create-coupon`. (named "create-coupon")
2. Set the request method to `POST`.
3. Copy the example JSON payload below and paste it into the request body.
4. Click the "Send" button to send the request to the server.
5. Verify that the response includes the newly created coupon in the Stripe connected account.

### Example JSON Payload
#### createProductAndPrice
`{
"name": "product",
"description": "A description of my product",
"unitAmount": 2000,
"accountId": "acct_1MllhCQ9p8STjU5P"
}`
#### create-coupon
`{
    "percentOff": 50,
    "duration": "once",
    "durationInMonths": 0,
    "productIds": [
        "prod_NVm6a9CFc2qGuj",
        "prod_NWqEOZBokjW7RS"
    ]
}`


> **(You can do a stripe login by using email and password that is in announcement channel in discord)**

![image](https://user-images.githubusercontent.com/85776877/225215442-f737e0fc-e2c8-4e15-8316-ea04954a94a2.png)
